types/index.d.ts

```diff
@@ -1330,6 +1330,7 @@ declare namespace Knex {
     wrap<TResult2 = TResult>(before: string, after: string): Raw<TResult>;
     toSQL(): Sql;
     queryContext(context: any): Raw<TResult>;
+    queryContext(): any;
```

Comment:
would it be better defined as Record<string, any>?

Reply:
I'm not sure, would that be for the return type?

From what I can tell it returns exactly what was put in when called with an argument, so I figured it would be best to just have those types match

Reply:
ok, fair :). does it work correctly if user passes e. g. string?

Reply:
Yep, this is the underlying implementation: https://github.com/knex/knex/blob/988fb243898d746a759d422762685a79eddf99ca/lib/helpers.js#L79-L85

And the docs say the queryContext is shallowly cloned when a query builder is cloned:
https://github.com/knex/knex/blob/f87b28c3e3a6e2624d83ab9c4c68ecffc913aee5/lib/query/builder.js#L93
which uses [`lodash.clone`](https://lodash.com/docs/4.17.15#clone) and should work on any passed in